src/main/scala/com/gu/zuora/ProductSubscriptionBuilders.scala

```diff
@@ -1,15 +1,34 @@
 package com.gu.zuora
 
+import com.gu.config.Configuration
 import com.gu.i18n.Country
-import com.gu.support.catalog.ProductRatePlanId
-import com.gu.support.config.ZuoraConfig
+import com.gu.support.catalog
+import com.gu.support.catalog.{Product, ProductRatePlan, ProductRatePlanId}
+import com.gu.support.config.{TouchPointEnvironments, ZuoraConfig}
 import com.gu.support.promotions.{PromoCode, PromotionService}
-import com.gu.support.workers.{Contribution, DigitalPack}
+import com.gu.support.workers.exceptions.CatalogDataNotFoundException
+import com.gu.support.workers.{Contribution, DigitalPack, ProductType}
 import com.gu.support.zuora.api._
 import org.joda.time.{DateTimeZone, LocalDate}
 
+import scala.util.{Failure, Success, Try}
+
 object ProductSubscriptionBuilders {
 
+  def getProductRatePlanId[PT <: ProductType, P <: Product](product: P, productType: PT): ProductRatePlanId = {
+    val touchpointEnvironment = TouchPointEnvironments.fromStage(Configuration.stage)
+    def getRatePlans[T <: Product](product: T): Seq[ProductRatePlan[Product]] = product.ratePlans.getOrElse(touchpointEnvironment, Nil)
+
+    val ratePlans: Seq[ProductRatePlan[Product]] = getRatePlans(catalog.DigitalPack)
+
+    val maybeProductRatePlanId: Option[ProductRatePlanId] = ratePlans.find(p => p.billingPeriod == productType.billingPeriod) map (_.id)
+
+    Try(maybeProductRatePlanId.get) match {
```

Comment:
Is there a reasons this needs a `try`?

Reply:
I want to move away from keeping `ProductRatePlanId`s among Zuora config here: https://github.com/guardian/support-libraries/blob/master/support-config/src/main/scala/com/gu/support/config/ZuoraConfigProvider.scala 

We're already maintaining a mapping as [here](https://github.com/guardian/support-libraries/blob/master/support-models/src/main/scala/com/gu/support/catalog/Product.scala), and I think these ids belong to a notion of the Zuora catalog rather than Zuora related config anyway. 

But the upside of keeping these ids among Zuora config is they were required config as soon as we tried to load the config, if there was a `ProductRatePlanId` missing that we needed this would fail fast. If we make a mistake and we're missing a `ProductRatePlanId` that should be defined, then when we try to `maybeProductRatePlanId` on line 24 will be a None. Then, `maybeProductRatePlanId.get` will throw `java.util.NoSuchElementException: None.get`. So, I wrapped it in a custom error so that if that happens we'll get a more descriptive error in the logs. 

But I'm not 100% sure this is the best way to handle this so v v open to ideas!

Reply:
Maybe you have some thoughts on this @jacobwinch ?

Reply:
I like the custom exception - it means I'd be able to understand a failure simply by looking at a failed execution in AWS Console without any further digging 💯👍